cloudSupport/src/main/scala/cromwell/cloudSupport/gcp/auth/GoogleAuthMode.scala

```diff
@@ -118,6 +120,23 @@ final case class ServiceAccountMode(override val name: String,
   }
 }
 
+final case class UserServiceAccountMode(override val name: String, scopes: java.util.List[String]) extends GoogleAuthMode {
+  import GoogleAuthMode.UserServiceAccountKey
+
+  private def extractServiceAccount(options: WorkflowOptions): String = {
+    options.get(UserServiceAccountKey) getOrElse {
+      throw new IllegalArgumentException(s"Missing parameters in workflow options: $UserServiceAccountKey")
+    }
+  }
+
+  override def credential(options: WorkflowOptions)(implicit as: ActorSystem, ec: ExecutionContext): Future[Credentials] = {
+    Future {
+      ServiceAccountCredentials.fromStream(new ByteArrayInputStream(extractServiceAccount(options).getBytes)).createScoped(scopes)
```

Comment:
Very very ToL:

A safer version of this specific line is `extractServiceAccount(options).getBytes("UTF-8")`, as `ServiceAccountCredentials` is not using the platform default encoding, but is specifying that the bytes [must be UTF-8](https://github.com/google/google-auth-library-java/blob/v0.7.0/oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java#L308-L309).

This line will work on 99.999% of systems. But every once in a while, you get someone who's set their default String encoding to something ~~weird~~ different. Generally I try to keep String<->Bytes conversions using the platform default encoding, or _always_ using UTF-8. This line mixes them.

That said, anyone who switches the default encoding on their system just might end up with a ton of other problems elsewhere in cromwell…